### PR TITLE
Add view for creating recipes

### PR DESCRIPTION
**Pull request by andreas-bauer**

### CODE REVIEW

**Comment by andreas-bauer** [comment]
The recipe edit and create view contains the basic UI elements and supports the mock API. The functionality with editing the amount of a victual is not implemented yet, because it will be more complicated with the different unit types, and therefore a separate PR.

File: ui/src/views/RecipeUpsert.vue [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+    this.availableVictuals = await this.victualApi.getAll();
+    if (this.editMode) {
+      this.recipe = await this.recipelApi.get(id);
+      this.recipe.ingredients.forEach(ing => this.moveToSelected(ing.victual));
+    }
+  }
+
+  async onCreate() {
+    this.recipelApi.create(this.recipe);
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:224]: We should await that Promise and stay on site if it fails

File: ui/src/views/RecipeUpsert.vue [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    this.$router.push("/recipes");
+  }
+
+  async onUpdate() {
+    this.recipelApi.update(this.recipe.id, this.recipe);
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:229]: await

File: ui/src/views/RecipeUpsert.vue [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    this.$router.push("/recipes");
+  }
+
+  moveToSelected(item) {
+    this.selectedVictuals.push(item);
+    const index = this.availableVictuals.indexOf(item);
+    this.availableVictuals.splice(index, 1);
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:236]: So you remove it on the list left? Not sure if we really have to do that? Is it possible to add the same victual twice?

File: ui/src/views/RecipeUpsert.vue [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                </v-toolbar>
+
+                <draggable
+                  class="list-group"
+                  :list="filteredVictuals"
+                  group="vic"
+                >
+                  <div v-for="item in filteredVictuals" :key="item.id">
+                    <v-divider></v-divider>
+                    <v-list-item>
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:77]: Can we do a `hand``symbol on mouse-over to make clear that it is draggable?

File: ui/src/views/RecipeUpsert.vue [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                  </v-btn>
+                </v-toolbar>
+                <draggable
+                  class="list-group"
+                  :list="selectedVictuals"
+                  group="vic"
+                >
+                  <div v-for="item in selectedVictuals" :key="item.id">
+                    <v-divider></v-divider>
+                    <v-list-item>
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:119]: Can we do a `hand``symbol on mouse-over to make clear that it is draggable / sortable?

**Comment by georg-schwarz** [comment]
Wow @andreas-bauer
You're a true frontend artist! Looks really nice. commented a few minor things, I can also resolve these if you want ;)

Good work!!

File: ui/src/views/RecipeUpsert.vue (hunk 3 above)
Comment (georg-schwarz) [reply ui/src/views/RecipeUpsert.vue:236]: Leave it like that for the time being

**Pull request merged**